symfony/plugins/orangehrmPimPlugin/Dao/EmpEmergencyContactDao.php

```diff
@@ -0,0 +1,179 @@
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Pim\Dao;
+
+use Exception;
+use InvalidArgumentException;
+use OrangeHRM\Core\Dao\BaseDao;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Entity\EmpEmergencyContact;
+use OrangeHRM\ORM\ListSorter;
+use OrangeHRM\ORM\Paginator;
+use OrangeHRM\Pim\Dto\EmpEmergencyContactSearchFilterParams;
+
+class EmpEmergencyContactDao extends BaseDao
+{
+
+    /**
+     * @param EmpEmergencyContact $empEmergencyContact
+     * @return EmpEmergencyContact
+     */
+    public function saveEmployeeEmergencyContact(EmpEmergencyContact $empEmergencyContact): EmpEmergencyContact
+    {
+        if ($empEmergencyContact->getSeqNo() === '0') {
+            $q = $this->createQueryBuilder(EmpEmergencyContact::class, 'eec');
+            $empNumber = $empEmergencyContact->getEmployee()->getEmpNumber();
+            $q->andWhere('eec.employee = :empNumber')
+                ->setParameter('empNumber', $empNumber);
+            $q->add('select', $q->expr()->max('eec.seqNo'));
+            $maxSeqNo = $q->getQuery()->getSingleScalarResult();
+            $seqNo = 1;
+            if (!is_null($maxSeqNo)) {
+                $seqNo += intval($maxSeqNo);
+            }
+            $empEmergencyContact->setSeqNo($seqNo);
+        }
+        $seqNo = intval($empEmergencyContact->getSeqNo());
+        if (!($seqNo < 100 && $seqNo > 0)) {
+            throw new InvalidArgumentException('Invalid `seqNo`');
+        }
+
+        $this->persist($empEmergencyContact);
+        return $empEmergencyContact;
+    }
+
+    /**
+     * Get Emergency contacts for given employee
+     * @param int $seqNo
+     * @param int $empNumber Employee Number
+     * @return EmpEmergencyContact|null EmpEmergencyContact objects as array
+     * @throws DaoException
+     */
+    public function getEmployeeEmergencyContact(int $empNumber, int $seqNo): ?EmpEmergencyContact
+    {
+        try {
+            $empEmergencyContact = $this->getRepository(EmpEmergencyContact::class)->findOneBy([
+                'employee' => $empNumber,
+                'seqNo' => $seqNo,
+            ]);
+            if ($empEmergencyContact instanceof EmpEmergencyContact) {
+                return $empEmergencyContact;
+            }
+            return null;
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * Delete Emergency contacts
+     * @param int $empNumber
+     * @param array|null $entriesToDelete
+     * @return int
+     * @throws DaoException
+     */
+    public function deleteEmployeeEmergencyContacts(int $empNumber, array $entriesToDelete): int
+    {
+        try {
+            $q = $this->createQueryBuilder(EmpEmergencyContact::class, 'ecc');
```

Comment:
what is the meaning of `ecc` alias, eec?

Reply:
ah sorry it should be eec, employee emergency contact